system/jlib/jlog.hpp

```diff
@@ -347,39 +385,71 @@ inline unsigned LogMsgFieldFromAbbrev(char const * abbrev)
     return 0;
 }
 
-// This function parses strings such as "AUD+CLS+DET+COD" and "STD+MIT-PID", and is used for fields attribute in XML handler descriptions
-
-inline unsigned LogMsgFieldsFromAbbrevs(char const * abbrevs)
+inline unsigned processAbbrevsString(char const * abbrevs, LogMsgAttType type)
 {
-    unsigned fields = 0;
+    unsigned values = 0;
     bool negate = false;
     bool more = true;
     while(more)
     {
-        if(strlen(abbrevs)<3) break;
-        unsigned field = LogMsgFieldFromAbbrev(abbrevs);
-        if(field)
+        if(strlen(abbrevs) < 3)
+            break;
+        unsigned value = 0;
+        switch(type)
+        {
+            case LogMsgAttType_Audience:
+                value = LogMsgAudFromAbbrev(abbrevs);
+                break;
+            case LogMsgAttType_fields:
+                value = LogMsgFieldFromAbbrev(abbrevs);
+                break;
+            case LogMsgAttType_class:
+                value = LogMsgClassFromAbbrev(abbrevs);
+                break;
+            default:
+                return 0;
+        }
+
+        if(value)
         {
             if(negate)
-                fields &= ~field;
+                values &= ~value;
             else
-                fields |= field;
+                values |= value;
         }
         switch(abbrevs[3])
         {
-        case '+':
-            negate = false;
-            abbrevs += 4;
-            break;
-        case '-':
-            negate = true;
-            abbrevs += 4;
-            break;
-        default:
-            more = false;
+            case '+':
+                negate = false;
+                abbrevs += 4;
+                break;
+            case '-':
+                negate = true;
+                abbrevs += 4;
+                break;
+            default:
+                more = false;
         }
     }
-    return fields;
+    return values;
+}
+
+// This function parses strings such as "ADT+PRO+USR" and "ALL+ADT-PRO"
+inline unsigned LogMsgAudsFromAbbrevs(char const * abbrevs)
+{
+    return processAbbrevsString(abbrevs, LogMsgAttType_Audience);
+}
+
+// This function parses strings such as "DIS+ERR+WRN+INF" and "ALL+PRO-INF"
+inline unsigned LogMsgClassesFromAbbrevs(char const * abbrevs)
+{
+    return processAbbrevsString(abbrevs, LogMsgAttType_class);
+}
+
+// This function parses strings such as "AUD+CLS+DET+COD" and "STD+MIT-PID", and is used for fields attribute in XML handler descriptions
+inline unsigned LogMsgFieldsFromAbbrevs(char const * abbrevs)
+{
+    return processAbbrevsString(abbrevs, LogMsgAttType_fields);
```

Comment:
trivial/not new: not sure any of these methods should be in the header, they are used internally to log.
Might be good to move to them (and thew new LogMsgAttType enumerated type) to jlog.cpp